### PR TITLE
Update linting recommendations for TypeScript

### LINKED ISSUE #91: TypeScript recipe suggests deprecated tslint
The [TypeScript recipe](https://github.com/microsoft/code-with-engineering-playbook/blob/master/pull-requests/code-reviews/recipes/TypeScript.md#linter) suggests `TSLint` as a linter. TSLint is being [deprecated](https://medium.com/palantir/tslint-in-2019-1a144c2317a9) in favor of improving ESLint's TypeScript support. These docs can be updated with some instructions on using ESLint to lint TypeScript code (e.g. https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/README.md).

I can open a PR with updates (tomorrow since it's EOD :smile:)

### PR DESCRIPTION
**Pull request by chrobles**
## Update Linting Recommendations for TypeScript

[TSLint is deprecated](https://medium.com/palantir/tslint-in-2019-1a144c2317a9). This PR updates guidance for configuring `eslint` & `prettier` for TypeScript projects, and adds setup info for autoformatting in VSCode.

Closes #91.

### CODE REVIEW

File: pull-requests/code-reviews/recipes/TypeScript.md [hunk 1]
```diff
@@ -1,92 +1,111 @@
...
+Per guidance outlined in [Palantir's 2019 TSLint roadmap](https://medium.com/palantir/tslint-in-2019-1a144c2317a9),
+TypeScript code should be linted with [ESLint](https://github.com/eslint/eslint). Resources for
+linting TypeScript code with ESLint can be found in the [typescript-eslint](https://github.com/typescript-eslint/typescript-eslint)
+repository.
+
+To [install and configure linting with ESLint](https://github.com/typescript-eslint/typescript-eslint/tree/master/docs/getting-started/linting),
+install the following dependencies:
+
+```bash
+npm i -D eslint @typescript-eslint/parser @typescript-eslint/eslint-plugin
```
Comment (davidkpiano) [inline pull-requests/code-reviews/recipes/TypeScript.md]:
```suggestion
# Install as a dev-dependency
npm install -D eslint @typescript-eslint/parser @typescript-eslint/eslint-plugin
```

File: pull-requests/code-reviews/recipes/TypeScript.md [hunk 2]
```diff
@@ -1,92 +1,111 @@
...
+module.exports = {
+  root: true,
+  parser: '@typescript-eslint/parser',
+  plugins: [
+    '@typescript-eslint',
+  ],
+  extends: [
+    'eslint:recommended',
+    'plugin:@typescript-eslint/eslint-recommended',
+    'plugin:@typescript-eslint/recommended',
```
Comment (davidkpiano) [inline pull-requests/code-reviews/recipes/TypeScript.md]: Why so many recommendations?

Comment (chrobles) [reply pull-requests/code-reviews/recipes/TypeScript.md]: This is the `eslintrc` recommended by [`typescript-eslint`'s getting started](https://github.com/typescript-eslint/typescript-eslint/tree/master/docs/getting-started/linting#configuration), welcome to suggestions here

File: pull-requests/code-reviews/recipes/TypeScript.md [hunk 3]
```diff
@@ -1,92 +1,111 @@
...
+};
+```
 
-All developers should run `TSLint` in a pre-commit hook to ensure standard formatting. We highly recommend using an editor integration like [vscode-tslint](https://github.com/Microsoft/vscode-tslint) to provide realtime feedback.
+Add a `.eslintignore` with files & folders to omit:
 
-Recommended: [lint-staged and husky](https://github.com/okonet/lint-staged#installation-and-setup)
+```ignore
+# don't ever lint node_modules
+node_modules
```
Comment (davidkpiano) [inline pull-requests/code-reviews/recipes/TypeScript.md]:
```suggestion
```

`node_modules` are never included in eslint

File: pull-requests/code-reviews/recipes/TypeScript.md [hunk 4]
```diff
@@ -1,92 +1,111 @@
...
+# don't lint build output
+dist
+# don't lint nyc coverage
+coverage
+```
```
Comment (davidkpiano) [inline pull-requests/code-reviews/recipes/TypeScript.md]: A better (default) solution might be to point the `--ignore-path` to the project's `.gitignore` file as explained here: https://eslint.org/docs/2.13.1/user-guide/configuring#using-an-alternate-file

**Comment by chrobles** [comment]
Thanks for the review @davidkpiano!

**Comment by debugerr** [comment]
Nice contribution thanks! Is there a way to enforce the linting in a build pipeline?

**Comment by chrobles** [comment]
Thanks for the review @fnocera! @debugerr, we've enforced linting in Azure Pipelines with a 
```yaml
- script: npm run lint
```
step.

It seems I don't have write access so I am unable to complete this PR myself :smile:

**Pull request merged**